src/main/java/org/opensearch/securityanalytics/threatIntel/transport/TransportGetIocFindingsAction.java

```diff
@@ -0,0 +1,143 @@
+        this.filterByEnabled = SecurityAnalyticsSettings.FILTER_BY_BACKEND_ROLES.get(this.settings);
+        this.clusterService.getClusterSettings().addSettingsUpdateConsumer(SecurityAnalyticsSettings.FILTER_BY_BACKEND_ROLES, this::setFilterByEnabled);
+    }
+
+    @Override
+    protected void doExecute(Task task, GetIocFindingsRequest request, ActionListener<GetIocFindingsResponse> actionListener) {
+        User user = readUserFromThreadContext(this.threadPool);
+
+        String validateBackendRoleMessage = validateUserBackendRoles(user, this.filterByEnabled);
+        if (!"".equals(validateBackendRoleMessage)) {
+            actionListener.onFailure(new OpenSearchStatusException("Do not have permissions to resource", RestStatus.FORBIDDEN));
+            return;
+        }
+        Table tableProp = request.getTable();
+        FieldSortBuilder sortBuilder = SortBuilders
+                .fieldSort(tableProp.getSortString())
+                .order(SortOrder.fromString(tableProp.getSortOrder()));
+        if (tableProp.getMissing() != null && !tableProp.getMissing().isBlank()) {
+            sortBuilder.missing(tableProp.getMissing());
+        }
+
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder()
+                .sort(sortBuilder)
+                .size(tableProp.getSize())
+                .from(tableProp.getStartIndex())
+                .fetchSource(new FetchSourceContext(true, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY))
+                .seqNoAndPrimaryTerm(true)
+                .version(true);
+
+        BoolQueryBuilder queryBuilder = QueryBuilders.boolQuery();
+        List<String> findingIds = request.getFindingIds();
+
+        if (findingIds != null && !findingIds.isEmpty()) {
+            queryBuilder.filter(QueryBuilders.termsQuery("id", findingIds));
+        }
+
+        Instant startTime = request.getStartTime();
+        Instant endTime = request.getEndTime();
+        if (startTime != null && endTime != null) {
+            long startTimeMillis = startTime.toEpochMilli();
+            long endTimeMillis = endTime.toEpochMilli();
+            QueryBuilder timeRangeQuery = QueryBuilders.rangeQuery("timestamp")
+                    .from(startTimeMillis) // Greater than or equal to start time
+                    .to(endTimeMillis); // Less than or equal to end time
+            queryBuilder.filter(timeRangeQuery);
+        }
+
+        if (tableProp.getSearchString() != null && !tableProp.getSearchString().isBlank()) {
+            queryBuilder.should(QueryBuilders
+                    .queryStringQuery(tableProp.getSearchString())
+            ).should(
+                    QueryBuilders.nestedQuery(
+                            "queries",
+                            QueryBuilders.boolQuery()
+                                    .must(
+                                            QueryBuilders
+                                                    .queryStringQuery(tableProp.getSearchString())
```

Comment:
Wondering if the way to go about the question in https://github.com/eirsep/security-analytics/pull/6#issuecomment-2179554895 is to supply the IOC value in the `searchString`? If so, this query builder would need to also search the finding `iocValue` field.